### ISSUE TITLE
MDS slave table catchup takes longer than expected

### ISSUE THREAD

**Issue by redlicha** [openvstorage/volumedriver#350]
dry run:
```
2017-09-06 01:05:40 784280 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MetaDataServerTable - 00000000000010c6 - info - catch_up: c8dac948-2d2a-46c5-8339-afa7c14b7249: request to catch up; dry run: DryRun::T, check scrub ID: CheckScrubId::T
[...]
2017-09-06 01:05:40 859580 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MetaDataStoreBuilder - 00000000000010d5 - info - update_metadata_store_: c8dac948-2d2a-46c5-8339-afa7c14b7249: determining the TLogs to replay
2017-09-06 01:05:40 859630 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/BackendRestartAccumulator - 00000000000010d6 - info - BackendRestartAccumulator: cork range: ( 1ebe523b-36cc-458e-9653-efc4f0e8282b,  8444abf6-1139-4aa2-95d2-3ebb1461be70]
2017-09-06 01:05:40 859874 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MDSMetaDataBackend - 00000000000010d7 - info - ~MDSMetaDataBackend: c8dac948-2d2a-46c5-8339-afa7c14b7249: used clusters: 5449744
```
actual catch up:
```
2017-09-06 01:05:40 861210 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MetaDataServerTable - 00000000000010dc - info - catch_up: c8dac948-2d2a-46c5-8339-afa7c14b7249: request to catch up; dry run: DryRun::F, check scrub ID: CheckScrubId::T
[...]
2017-09-06 01:05:40 940654 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MetaDataStoreBuilder - 00000000000010eb - info - update_metadata_store_: c8dac948-2d2a-46c5-8339-afa7c14b7249: determining the TLogs to replay
2017-09-06 01:05:40 940682 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/BackendRestartAccumulator - 00000000000010ec - info - BackendRestartAccumulator: cork range: ( 1ebe523b-36cc-458e-9653-efc4f0e8282b,  8444abf6-1139-4aa2-95d2-3ebb1461be70]
2017-09-06 01:05:40 940757 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MetaDataStoreBuilder - 00000000000010ed - info - update_metadata_store_: c8dac948-2d2a-46c5-8339-afa7c14b7249: replaying 2 TLogs
2017-09-06 01:05:40 940825 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000010ee - info - Logger: Entering read c8dac948-2d2a-46c5-8339-afa7c14b7249 tlog_8c0fdf4e-8754-4753-af2b-540f3ece3eb8
2017-09-06 01:05:40 945140 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000010ef - info - ~Logger: Exiting read for c8dac948-2d2a-46c5-8339-afa7c14b7249 tlog_8c0fdf4e-8754-4753-af2b-540f3ece3eb8
2017-09-06 01:05:40 945178 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000010f0 - info - ~Logger: Exiting read for c8dac948-2d2a-46c5-8339-afa7c14b7249
2017-09-06 01:05:40 946290 -0700 - SV4SRV0007 - 11519/0x00007f1c15ff4700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000010f1 - info - Logger: Entering read c8dac948-2d2a-46c5-8339-afa7c14b7249 tlog_8444abf6-1139-4aa2-95d2-3ebb1461be70
2017-09-06 01:05:40 950796 -0700 - SV4SRV0007 - 11519/0x00007f1c15ff4700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000010f2 - info - ~Logger: Exiting read for c8dac948-2d2a-46c5-8339-afa7c14b7249 tlog_8444abf6-1139-4aa2-95d2-3ebb1461be70
2017-09-06 01:05:40 950838 -0700 - SV4SRV0007 - 11519/0x00007f1c15ff4700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000010f3 - info - ~Logger: Exiting read for c8dac948-2d2a-46c5-8339-afa7c14b7249
[...]
2017-09-06 01:05:57 619699 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/CachedMetaDataStore - 00000000000011bc - info - processPages: c8dac948-2d2a-46c5-8339-afa7c14b7249: starting processing pages for cloneID 0
2017-09-06 01:05:57 632572 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/CachedMetaDataStore - 00000000000011bd - info - processPages: c8dac948-2d2a-46c5-8339-afa7c14b7249: finished. 16384 entries written in 530 pages.
[...]
2017-09-06 01:06:00 884090 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MDSMetaDataBackend - 00000000000011c7 - info - setCork: c8dac948-2d2a-46c5-8339-afa7c14b7249: 8444abf6-1139-4aa2-95d2-3ebb1461be70
[...]
2017-09-06 01:06:02 745407 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MDSMetaDataBackend - 0000000000001216 - info - set_scrub_id: c8dac948-2d2a-46c5-8339-afa7c14b7249: setting scrub ID ddc04a72-a13d-490d-a60a-22f211086cb8
2017-09-06 01:06:02 747336 -0700 - SV4SRV0007 - 11519/0x00007f1c70996700 - volumedriverfs/MDSMetaDataBackend - 0000000000001225 - info - ~MDSMetaDataBackend: c8dac948-2d2a-46c5-8339-afa7c14b7249: used clusters: 5466095
```

A similar behaviour is observed when reconfiguring the volume to use the slave:
```
2017-09-06 04:30:39 485081 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/XMLRPCTimingWrapper - 00000000000213ba - info - execute: Arguments for updateMetaDataBackendConfig are {[metadata_backend_config:FgAAAAAAAABzZXJpYWxpemF0aW9uOjphcmNoaXZlDAAECAQIAQAAAAAAAAAAAQAmAAAAAAAA
AHZvbHVtZWRyaXZlcjo6TURTTWV0YURhdGFCYWNrZW5kQ29uZmlnAQMAAAAAAAAAAAAAAAAB
AAAAAAAAAAEAAAAAAQAAAAwAAAAAAAAAMTcyLjI2LjE2LjEwvWYBAAAAFAAAAAAAAAA=,volume_id:c8dac948-2d2a-46c5-8339-afa7c14b7249,vrouter_cluster_id:611f7b47-ad68-4d49-ae74-d18aef6dd412]}
[...]
2017-09-06 04:30:39 736253 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/MetaDataStoreBuilder - 00000000000213e1 - info - update_metadata_store_: c8dac948-2d2a-46c5-8339-afa7c14b7249: replaying 1 TLogs
2017-09-06 04:30:39 736337 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000213e2 - info - Logger: Entering read c8dac948-2d2a-46c5-8339-afa7c14b7249 tlog_32f963a5-1c70-4b62-ad95-7ff9c3d33487
2017-09-06 04:30:39 746189 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000213e3 - info - ~Logger: Exiting read for c8dac948-2d2a-46c5-8339-afa7c14b7249 tlog_32f963a5-1c70-4b62-ad95-7ff9c3d33487
2017-09-06 04:30:39 746238 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/BackendConnectionInterfaceLogger - 00000000000213e4 - info - ~Logger: Exiting read for c8dac948-2d2a-46c5-8339-afa7c14b7249
[...]
2017-09-06 04:31:13 571852 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/CachedMetaDataStore - 0000000000021433 - info - processPages: c8dac948-2d2a-46c5-8339-afa7c14b7249: starting processing pages for cloneID 0
[...]
2017-09-06 04:31:13 578655 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/CachedMetaDataStore - 0000000000021434 - info - processPages: c8dac948-2d2a-46c5-8339-afa7c14b7249: finished. 8192 entries written in 266 pages.
2017-09-06 04:31:21 741605 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/MDSMetaDataBackend - 000000000002144f - info - setCork: c8dac948-2d2a-46c5-8339-afa7c14b7249: 32f963a5-1c70-4b62-ad95-7ff9c3d33487
[...]
2017-09-06 04:31:21 865602 -0700 - SV4SRV0007 - 11519/0x00007f1be77fe700 - volumedriverfs/XMLRPCTimingWrapper - 0000000000021470 - info - execute: Call updateMetaDataBackendConfig took 42.3805 seconds

```

**Comment by redlicha** [comment]
Observed consistently for the given volume but not for other volumes on that instance. Complete log set aside on the loghost under redlicha/gh-350.

**Comment by redlicha** [comment]
RocksDB writes stalled due to rocksdb threads being busy with compaction, cf. https://github.com/facebook/rocksdb/wiki/RocksDB-Basics#avoiding-stalls ?

**Comment by redlicha** [comment]
While RocksDB write stalls during compaction still should be addressed (-> #364 ), the cause of this is most likely #363 .

**Comment by redlicha** [comment]
Reopen if #363 doesn't solve it.
